### PR TITLE
FieldDefinition: Add missing forbidden names

### PR DESCRIPTION
**Pull request by blankse**
`Fatal error: Declaration of Pimcore\Model\DataObject\Test::setClassId(?string $classId) must be compatible with Pimcore\Model\DataObject\Concrete::setClassId($o_classId) in /var/www/html/var/classes/DataObject/Test.php on line 188`

See also https://github.com/pimcore/pimcore/pull/17109

### CODE REVIEW

File: public/js/pimcore/object/classes/data/data.js [hunk 1]
```diff
@@ -25,7 +25,7 @@ pimcore.object.classes.data.data = Class.create({
         "versions", "properties", "permissions", "permissionsforuser", "childamount", "apipluginbroker", "resource",
         "parentClass", "definition", "locked", "language", "omitmandatorycheck", "idpath", "object", "fieldname",
         "property", "localizedfields", "parentid", "scheduledtasks", "latestVersion", "haschildren", "siblings", "hassiblings",
-        "childrenSortby", "childrensortorder", "versioncount", "dirtylanguages", "dirtyfields", "classtitle"
+        "childrensortby", "childrensortorder", "versioncount", "dirtylanguages", "dirtyfields", "classtitle", "classid"
```
Comment (kingjia90) [inline public/js/pimcore/object/classes/data/data.js]: Nice catch, LGTM but i've noticed that it breaks for every other getter as well eg. `valueForFieldName`, `dao`, `NextParentForInheritance` ,`ClosestParentOfClass`, `LazyLoadedFieldNames`, `RelationData`

Could you please add these as well? TIA

image
image
image

Comment (blankse) [reply public/js/pimcore/object/classes/data/data.js]: @kingjia90 I added it. I also sorted it alphabetically. So we see faster missing properties.

Comment (kingjia90) [reply public/js/pimcore/object/classes/data/data.js]: Thank you for the changes and sorting, that's helpful.

Indeed, it's a little hard to read and manage such a long array, and we can't break a line for each element.
I am having a sort of brain-fart, what do you think about a "counter-intuitive" (just for human/dev readability) solution to make it an associative array, group by similar domain/areas and that gets flattened for the compiler? 🙈

Comment (blankse) [reply public/js/pimcore/object/classes/data/data.js]: @kingjia90 I don't think I quite follow you. Do you mean to make a multidimensional list out of it?

Maby we generate the list via Ajax call? Via PHP reflection?

We can write it down as an issue. But I think it's too much for this PR.

Comment (kingjia90) [reply public/js/pimcore/object/classes/data/data.js]: Yes, some kind of that, like
```
someconst: {
  'parent': ["parent", "parentclass", "parentid"],
  'id': ["id", "byid"]
}
```
and flatten/reduced it to be like `"parent", "parentclass", "parentid","id", "byid"`

but agree, that's not worth it to invest so much time on that.

So far LGTM as is, i will have a deeper review and merge soon or latest on Monday. Wish you a nice weekend!

Comment (blankse) [reply public/js/pimcore/object/classes/data/data.js]: @kingjia90 I wish you also a nice weekend!

**Comment by robertSt7** [comment]
@blankse Thanks a lot

**Pull request merged**